command/recorderd/job_calendar.go

```diff
@@ -0,0 +1,515 @@
+func isTimeDataFirstEarlierThanSecond(first TimeData, second TimeData) bool {
+	if first.hour < second.hour {
+		return true
+	}
+	if first.minute < second.minute {
+		return true
+	}
+	return false
+}
+
+func (j *JobCalendarData) isTimeBooked(event time.Time) bool {
+	weekDay := event.Weekday()
+	events := j.events[weekDay]
+	for _, t := range events {
+		afterOrEqualToStartTime := t.start.Before(event) || t.start.Equal(event)
+		beforeOrEqualToEndTime := t.stop.After(event) || t.stop.Equal(event)
+		if afterOrEqualToStartTime && beforeOrEqualToEndTime {
+			return true
+		}
+	}
+	return false
+}
+
+func (j *JobCalendarData) pickInitialiseStartEvent(event time.Time) interface{} {
+	if j.isTimeBooked(event) {
+		return event.Add(delayOfStartStop)
+	}
+	return j.pickNextStartEvent(event)
+}
+
+func (j *JobCalendarData) pickInitialiseStopEvent(event time.Time) interface{} {
+	if j.isTimeBooked(event) {
+		return j.pickNextStopEvent(event)
+	}
+	return event.Add(delayOfStartStop)
+}
+
+func (j *JobCalendarData) pickNextStartEvent(event time.Time) interface{} {
+	for _, e := range j.flattenEvents.start {
+		if e.After(event) {
+			return e
+		}
+	}
+	return nil
+}
+
+func (j *JobCalendarData) pickNextStopEvent(event time.Time) interface{} {
+	for _, e := range j.flattenEvents.stop {
+		if e.After(event) {
+			return e
+		}
+	}
+	return nil
+}
+
+func (j *JobCalendarData) rescheduleStartEventsPrior(event time.Time) {
+	if 0 == len(j.flattenEvents.start) || j.flattenEvents.start[0].After(event) {
+		return
+	}
+	times := j.flattenEvents.start
+	newSlices := make([]time.Time, 0, len(times))
+	schedules := make([]time.Time, 0, len(times))
+	for i, t := range times {
+		if t.Before(event) || t.Equal(event) {
+			schedules = append(schedules, t.Add(oneWeekDuration))
+		} else {
+			newSlices = append(newSlices, times[i:]...)
+			break
+		}
+	}
+	newSlices = append(newSlices, schedules...)
+	j.flattenEvents.start = newSlices
+}
+
+func (j *JobCalendarData) rescheduleStopEventsPrior(event time.Time) {
+	if 0 == len(j.flattenEvents.stop) || j.flattenEvents.stop[0].After(event) {
+		return
+	}
+	times := j.flattenEvents.stop
+	newSlices := make([]time.Time, 0, len(times))
+	schedules := make([]time.Time, 0, len(times))
+	for i, t := range times {
+		if t.Before(event) || t.Equal(event) {
+			schedules = append(schedules, t.Add(oneWeekDuration))
+		} else {
+			newSlices = append(newSlices, times[i:]...)
+			break
+		}
+	}
+	newSlices = append(newSlices, schedules...)
+	j.flattenEvents.stop = newSlices
+}
+
+func (j *JobCalendarData) removeEventFrom(times []time.Time, event time.Time) ([]time.Time, error) {
+	exist, idx := isEventAlreadyExist(times, event)
+	if !exist {
+		return times, nil
+	}
+	return append(times[:idx], times[idx+1:]...), nil
+}
+
+func (j *JobCalendarData) weekDayCurrent2Target(current time.Weekday, target time.Weekday) int {
+	return int(target) - int(current)
+}
+
+func (j *JobCalendarData) parseClockStr(clock string) (TimeData, error) {
```

Comment:
Did you consider `t, _ := time.Parse(time.Kitchen, "3:04PM")`?

Reply:
I can't find layout fits current "18:00", `time.Kitchen` is "6:00PM"

Reply:
https://gobyexample.com/time-formatting-parsing
`t, _ := time.Parse("15:04", clockStr)`

Reply:
ok, updated